<?xml version="1.0"?>
<launch>

<arg name="port" value="/dev/ttyUSB0"/>
<arg name="baudrate" value="1000000"/>
<!-- Dynamixel's ID of each joint -->
<arg name="lhand" value="1"/>
<arg name="rhand" value="6"/>
<arg name="head" value="5"/>
<arg name="neck" value="4"/>

<!-- <arg name="body_language_emotion_topic" value="/face_emotions"/>
<arg name="body_language_emotion_msg_type" value="FaceEmotions"/> -->
<arg name="body_language_emotion_topic" value="/manual_emotions"/>
<arg name="body_language_emotion_msg_type" value="String"/>


<!-- Emotion interactions node -->
    <!-- <include file="$(find infrastructure)/launch/initiate_facial.launch"/> -->
<!-- Gaze_pose launch -->
    <!-- <include file="$(find infrastructure)/launch/gaze_pose.launch"/> -->
<!-- Speech_to_text launch -->
    <!-- <include file="$(find infrastructure)/launch/speech_to_text.launch"/> -->
<!-- Text_to_speech launch -->
    <!-- <include file="$(find infrastructure)/launch/text_to_speech.launch"/> -->
<!-- Speech_emotion_analysis launch-->
    <!-- <include file="$(find infrastructure)/launch/speech_emotion_analysis.launch"/> -->

<!-- Controlling the Dynamixels -->
    <node pkg="infrastructure" type="DynaControl.py" name="Dynamixels_controller" output="screen" 
        args="--port $(arg port) 
                --baudrate $(arg baudrate) 
                --lhand $(arg lhand) 
                --rhand $(arg rhand) 
                --head $(arg head) 
                --neck $(arg neck)"/>
                
    <node pkg="infrastructure" type="body_language_controller.py" name="body_language_controller" output="screen"
        args="--topic $(arg body_language_emotion_topic) 
                --msg_type $(arg body_language_emotion_msg_type)"/>
    


<!-- Bridging between ROS and HTML interface node -->
    <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch"/>

<!-- Changing rosbridge node's topic name from default to "web_interface" -->
    <remap from="rosbridge_websocket" to="web_interface"/>




</launch>